overlord/snapstate/snapstate.go

```diff
@@ -988,8 +990,9 @@ func Switch(st *state.State, name, channel string) (*state.TaskSet, error) {
 	}
 
 	snapsup := &SnapSetup{
-		SideInfo: snapst.CurrentSideInfo(),
-		Channel:  channel,
+		SideInfo:    snapst.CurrentSideInfo(),
+		Channel:     channel,
+		InstanceKey: snapst.InstanceKey,
```

Comment:
is this change tested? I see tests for most changes here, but not this one

Reply:
Added a run through test for switch.